app/models/host_aspect_base.rb

```diff
@@ -0,0 +1,25 @@
+class HostAspectBase < ActiveRecord::Base
```

Comment:
That would work. Also: I'm partial to not naming this abstraction an "aspect" as it can be confusing (traditionally aspect is an abstraction from aspect-oriented programming, which is rather common, if not in ruby world). The name of this pattern from the "gang of four" book is facet, so we could use that.